sql: handle inverted index queries involving NULL

justinj opened the pull request:
I will plan on backporting this to 2.0.

Fixes #24244.

Release note (bug fix): inverted index queries involving NULL are now
properly handled.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/24251)
<!-- Reviewable:end -->

RaduBerinde commented:
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[pkg/sql/opt/idxconstraint/index_constraints.go, line 864 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/24251#-L8blhN15PHA4VKdqQhC:-L8blhN2lfbRpSn08XIJ:b9vzzbs) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf7af4f0c4adf2cb2a6e831aaa6ddb0cb3220efe/pkg/sql/opt/idxconstraint/index_constraints.go#L864)):*
> ```Go
> 
> 		// NULL is never contained, return no spans.
> 		if rightDatum == tree.DNull {
> ```

This changed on `master`, will be something like
```
{
  c.contradiction(0 /* offset */, out)
  return true
}
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/24251)*
<!-- Sent from Reviewable.io -->

Commit:
sql: handle inverted index queries involving NULL

I will plan on backporting this to 2.0.

Fixes #24244.

Release note (bug fix): inverted index queries involving NULL are now
properly handled.

justinj commented:
Whoops, fixed!

justinj commented:
ping in case this slipped through the cracks, would like to cherry-pick for 2.0.1.

RaduBerinde commented:
Indeed it did, sorry about that. LGTM

justinj commented:
no worries!

bors r=RaduBerinde

justinj commented:
trying again

bors r=RaduBerinde
